unit_tests/test_events.py

```diff
@@ -274,22 +275,42 @@ def test_default_filters(self):
                                                                  line_number=22,
                                                                  line="other back trace that shouldn't be filtered")
 
-        self.wait_for_event_log_change(log_content_before)
-
-        log_content_after = self.get_event_logs()
+        log_content_after = self.wait_for_event_log_change('events.log', log_content_before)
         self.assertIn('other back trace', log_content_after)
         self.assertNotIn('supressed', log_content_after)
 
     def test_failed_stall_during_filter(self):
-
-        with DbEventsFilter(type="NO_SPACE_ERROR"), DbEventsFilter(type='BACKTRACE', line='No space left on device'):
-
+        with DbEventsFilter(type="NO_SPACE_ERROR"), \
+                DbEventsFilter(type='BACKTRACE', line='No space left on device'):
             event = DatabaseLogEvent(type="REACTOR_STALLED", regex="B")
             event.add_info_and_publish(node="A", line_number=22,
                                        line="[99.80.124.204] [stdout] Mar 31 09:08:10 warning|  reactor stall 20")
             logging.info(event.severity)
             self.assertTrue(event.severity == Severity.CRITICAL)
 
+    @unittest.skip("for manual use only")
+    def test_measure_speed_of_events_processing(self):  # pylint: disable=no-self-use
+        non_guaranteed_delivery = []
+        for _ in range(5):
+            start_time = time.time()
+            for _ in range(1000):
+                evt = DatabaseLogEvent(type="NO_SPACE_ERROR", regex="B")
+                evt.add_info(node="B", line_number=22, line="not filtered")
+                evt.publish(guaranteed=False)
+            non_guaranteed_delivery.append(time.time() - start_time)
+        guaranteed_delivery = []
+        for _ in range(5):
+            start_time = time.time()
+            for _ in range(1000):
+                evt = DatabaseLogEvent(type="NO_SPACE_ERROR", regex="B")
+                evt.add_info(node="B", line_number=22, line="not filtered")
+                evt.publish(guaranteed=True)
+            guaranteed_delivery.append(time.time() - start_time)
+        non_guaranteed_avg = sum(non_guaranteed_delivery) / len(non_guaranteed_delivery)
+        guaranteed_avg = sum(guaranteed_delivery) / len(guaranteed_delivery)
+        print(
```

Comment:
Just of of interest can you share the numbers from this test ?

Reply:
Non guranteed publishing of 1000 messages - 12.512795209884644
Guranteed publishing of 1000 messages - 13.74399471282959

Reply:
I'm more then happy with those numbers 😉